server/opts.go

```diff
@@ -261,6 +261,12 @@ type Options struct {
 	// and used as a filter criteria for some system requests
 	Tags jwt.TagList `json:"-"`
 
+	// StateDir is a directory to hold state of the server.
+	StateDir string `json:"-"`
```

Comment:
Let's require store_dir opt and tuck under own directory under there like `jetstream`

Reply:
Removed caching the files into a local directory for now, will have a different branch to address this part.